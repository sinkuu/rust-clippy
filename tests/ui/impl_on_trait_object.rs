#![allow(unused)]

trait Foo {
    fn bar(&self) {}
}

impl Foo {
    fn bar(&self) {}
}

trait Bar<T> {
    fn foo(&self) {}
}

impl<T> Bar<T> + Sync {
    fn foo(&self) {}
}

trait FooExtensions {
    fn bar(&self);
}

impl<'a> FooExtensions for Foo + Sync + 'a {
    fn bar(&self) {}
}

fn main() {}
