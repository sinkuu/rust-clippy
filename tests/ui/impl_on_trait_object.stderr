error: this adds a method to a trait object type, which is rarely intended
 --> $DIR/impl_on_trait_object.rs:7:1
  |
7 | impl Foo {
  | ^^^^^^^^ help: consider a blanket impl instead: `impl<T> Foo for T`
  |
  = note: `-D impl-on-trait-object` implied by `-D warnings`

error: this adds a method to a trait object type, which is rarely intended
  --> $DIR/impl_on_trait_object.rs:15:1
   |
15 | impl<T> Bar<T> + Sync {
   | ^^^^^^^^^^^^^^^^^^^^^ help: consider a blanket impl instead: `impl<T, U> Bar<T> for U`

error: this adds a method to a trait object type, which is rarely intended
  --> $DIR/impl_on_trait_object.rs:23:1
   |
23 | impl<'a> FooExtensions for Foo + Sync + 'a {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider a blanket impl instead: `impl<'a, T: Foo + Sync + ?Sized> FooExtensions for T`

